### ISSUE TITLE
Show unused var warning for unused re-frame reg-* handlers

### ISSUE THREAD

**Issue by dannyfreeman** [clojure-lsp/clojure-lsp#691]
**Is your feature request related to a problem? Please describe.**
Clojure-lsp does a great job of showing unused var warnings for `def`s and `defn`s. I use this feature to navigate to these warnings in my editor and delete the unused code. I am unable to do that with unused re-frame handlers. Clojure-lsp will show that there are 0 references to something like a `reg-sub` keyword, and if I happen to see that I can know to delete it. I have no way to jump around my codebase directly to these unused handlers with something like `lsp-treemacs-error-list` in emacs.

**Describe the solution you'd like**
I would like to see keywords registered with re-frame's `reg-sub`, `reg-event-db`, `reg-event-fx`, etc with `0` references to these keywords reported as unused-var warnings (or something similar to unused-var).
 
**Describe alternatives you've considered**
I've tried using clj-kondo directly to find unused re-frame handlers.
Both this repo: https://github.com/yannvanhalewyn/analyse-re-frame-usage-with-clj-kondo
and this gist https://gist.github.com/roman01la/c6a2e4db8d74f89789292002794a7142
are referenced in the [clj-kondo documentation](https://github.com/clj-kondo/clj-kondo/blob/master/doc/hooks.md#finding-unused-and-undefined-re-frame-subscriptions), but those examples have not worked, either reporting false positives or nothing at all.

**Additional context**
N/A

**Comment by ericdallo** [comment]
As we discussed on clojurians slack, this happens because clojure-lsp only check kondo `:var-definitions` to during the custom linter unused-public-vars, and `re-frame.core/reg-*` only register `keywords`.

I think we should fix this on clojure-lsp, but I wonder if we should register `var-definitions` for those like we do with potemkin macros, WDYT @borkdude?

**Comment by borkdude** [comment]
clj-kondo exposes the information needed to do so in clojure-lsp, but I don't think these should be treated as vars, but more like its own thing. You get a lot of information now with `{:context true}` or `{:context :re-frame.core}`

**Comment by ericdallo** [comment]
Yes, I thought about that, I'll try to implement with only the context info and let you know, thanks!

**Comment by ericdallo** [comment]
@dannyfreeman Done on master! It'd be great if you could test it from master, to do that you can clone and just run `make` to generate a local `clojure-lsp`

**Comment by dannyfreeman** [comment]
@ericdallo I just tested it out. It seems to be working great for me. Flycheck shows them and lets me jump to them when cycling through errors, the lsp-treemacs-errors-list command in lsp-mode for emacs recognizes them. 
I checked the diagnostics from the command line as well with

```
../../dev/clojure-lsp/clojure-lsp diagnostics | grep 'Unused public keyword' | wc -l
```

and it turned up 187 warnings, which sounds about right for the large re-frame app I work on.

Thank you very much for taking the time to implement this. This new information will really help me out.
